File: go/etl/etl.go
package etl

import "strings"

func Transform(in map[int][]string) (out map[string]int) {
	out = make(map[string]int)

	for score, letters := range in {
		for _, letter := range letters {
			out[strings.ToLower(letter)] = score
		}
	}

	return out

}

File: clojure/etl/src/etl.clj
(ns etl)

(defn partial-transform [[num letters]]
  (let [letters (map clojure.string/lower-case letters)]
    (apply merge (for [letter letters] (assoc {} letter num))))
  )

(defn transform [source]
  (->> (into [] source)
       (map partial-transform)
       (apply merge))
  )
